recommence/Checkpoint.py

```diff
@@ -58,6 +63,14 @@ def save(self) -> None:
             target=self._c.save_path,
         )
 
+    def _save_every(self) -> None:
+        def _save_every_thread(self) -> None:
+            while True:
+                self.save()
+                time.sleep(self._c.save_every_interval)
+        thread = Thread(target=_save_every_thread, args=(self,), daemon=True)
+        thread.start()
```

Comment:
We probably need to clean this thread up when we are done with it. At least in the `remove(self)` method, but also probably at exit. We can use the built-in python module for that:
```python
import atexit

atexit.register(cleanup_method)
```